internal/core/src/index/IndexFactory.cpp

```diff
@@ -255,32 +255,25 @@ IndexFactory::CreateScalarIndex(const CreateIndexInfo& create_index_info,
     switch (data_type) {
         // create scalar index
         case DataType::BOOL:
-            return CreateScalarIndex<bool>(
-                index_type, file_manager, space, data_type);
+            return CreateScalarIndex<bool>(index_type, file_manager, space);
         case DataType::INT8:
-            return CreateScalarIndex<int8_t>(
-                index_type, file_manager, space, data_type);
+            return CreateScalarIndex<int8_t>(index_type, file_manager, space);
         case DataType::INT16:
-            return CreateScalarIndex<int16_t>(
-                index_type, file_manager, space, data_type);
+            return CreateScalarIndex<int16_t>(index_type, file_manager, space);
         case DataType::INT32:
-            return CreateScalarIndex<int32_t>(
-                index_type, file_manager, space, data_type);
+            return CreateScalarIndex<int32_t>(index_type, file_manager, space);
         case DataType::INT64:
-            return CreateScalarIndex<int64_t>(
-                index_type, file_manager, space, data_type);
+            return CreateScalarIndex<int64_t>(index_type, file_manager, space);
         case DataType::FLOAT:
-            return CreateScalarIndex<float>(
-                index_type, file_manager, space, data_type);
+            return CreateScalarIndex<float>(index_type, file_manager, space);
         case DataType::DOUBLE:
-            return CreateScalarIndex<double>(
-                index_type, file_manager, space, data_type);
+            return CreateScalarIndex<double>(index_type, file_manager, space);
 
             // create string index
         case DataType::STRING:
         case DataType::VARCHAR:
             return CreateScalarIndex<std::string>(
-                index_type, file_manager, space, data_type);
+                index_type, file_manager, space);
```

Comment:
V2 not implement array

Reply:
We already depreacted the storage-v2-related api.